### ISSUE TITLE
Like Stack Strings but Not

### ISSUE THREAD

**Issue by utkonos** [MBCProject/mbc-markdown#64]
This behavior is very very similar to stack strings, but does not use the stack. A string is built character by character, out of order, and written to a variable in the `.data` section.

Here is an example in a packer I'm researching.

<img width="552" alt="image" src="https://user-images.githubusercontent.com/2775739/202553929-863ff84c-a8bb-4463-a705-bfddc7e39993.png">

The `lpProcName` parameter to `GetProcAddress` is pushed to the stack. Then the string is constructed letter by letter before the call to `GetProcAddress`.

Does this need a specific method? Is there a method that would include this and stack strings together? Should stack strings be bent to include this?

**Comment by utkonos** [comment]
Is the above "Variable Recomposition" B0012.004?

If so, I would also propose moving B0012.004 to Executable Code Obfuscation because it doesn't cause a problem with the disassembler it just makes analysis of the resulting correct assembly more difficult.

**Comment by dzbeck** [comment]
I think what you describe does fit under Variable Recomposition. It and some of the other methods (e.g., Argument Obfuscation) are under the Disassembler Evasion behavior because they can cause problems for linear disassemblers. Does that seem reasonable, or do you still think those methods belong under Executable Code Obfuscation?

**Comment by utkonos** [comment]
I understand. Is there a sample example with an address of the instructions that would do this available?

This prompted a few thoughts:

1. Taking this example above of variable recomposition: this doesn't cause a problem for the disassember. So, it fits in Executable Code Obfuscation. Should methods exist in two places?
2. For this thought, lets take it as a given that a method exists in two places. Does one get a different, more specific name?
3. Thinking from the perspective of detection, can there be identified some bytes at a closer resolution that can be identified as the bytes that cause the disassembly problem itself. These bytes could be smaller and more specific than the whole recomposition. This third one is why I would love to dig into a sample that has this behavior.

Here is an example YARA rule that matches the variable recomposition in the image above.

```yara
rule Obfuscated_VirtualProtect_String : MBC_B0032
{
    meta:
        author = "Malware Utkonos"
        date = "2022-10-17"
        description = "VirtualProtect moved byte by byte to location in data segment."
    strings:
        $vp = /(\xc6\x05.{5}){15}/
        $v = { c605[4]56 } private
    condition:
        uint16(0) == 0x5A4D and uint32(uint32(0x3C)) == 0x00004550 and
        $v in (@vp..@vp+!vp)
}
```

**Comment by dzbeck** [comment]
Unfortunately, I don't have a sample example (some methods evolved from vocabularies in MAEC rather than samples). 

Good question about whether the same method name could be defined under multiple behaviors. MBC models itself on ATT&CK and methods are sort of like sub-techniques - ATT&CK doesn't allow duplication of sub-technique names, so to date, MBC hasn't allowed duplication of method names, but it could...

The YARA rule is interesting to see. We're adding a "Detection" section to behavior pages - this would be a good addition once we decide where Variable Recomposition belongs. I'll check in with a couple of colleagues to see what they think of placement.

**Comment by dzbeck** [comment]
Hi @utkonos we've discussed it and decided to go with your recommendation and move both Argument Obfuscation and Variable Recomposition from Disassember Evasion to Executable Code Obfuscation. You can see the update in the mbc-utkonos branch (soon to be merged into master). Thanks again for your feedback!

**Comment by dzbeck** [comment]
see PR #73

**Comment by dzbeck** [comment]
addressed by PR #73
